### ISSUE TITLE
iOS - Wire up UI to data migration functions

### ISSUE THREAD

**Issue by dvdchr** [wordpress-mobile/WordPress-iOS#19668]
Hook up the data migration functionality to kickstart the process. Potential wiring areas:

### Export scenario

- Start the export process after the user taps the `Try the new Jetpack app`, and before they get redirected to JP. It doesn't matter if JP hasn't been installed since the exported data will be accessible to JP after it is installed.
  - There needs to be some validation/check that ensures that any local post drafts are synced. The implementation for this is covered in https://github.com/wordpress-mobile/WordPress-iOS/issues/19631.
- **In addition, also export the data during WP app launch.** This way we can broaden the reach of the migration flow from "Users tapping the 'Try the new Jetpack app'" to "Users opening the WP app at least once".
  - The export process during app launch will only be performed once if it's successful. Otherwise, we will try to export again during the next app launch.
  - Note that for the export process to start, the user must be logged in and has at least one blog.
- After the export process completes, WP will create a flag and store it in the shared UserDefaults area that JP can access. Once this flag is set, JP can read this flag and start the migration flow without relying on WP deep linking to JP.

### Import scenario

- When JP first launches, ensure that these conditions are fulfilled before starting the migration flow:
  - has the content migration flag enabled, AND
  - is not logged in, AND
  - has the "migration-ready flag" set from WP.
- Start the import process after verifying that the above conditions are fulfilled. 
  - Ensure the import process is completed BEFORE the first migration screen is shown.
  - By the time the welcome screen is shown, the user should've already been logged in. (`AccountHelper.isLoggedIn == true`).
- The wiring of the second migration screen — Notification — is out of scope from this task and is handled in https://github.com/wordpress-mobile/WordPress-iOS/issues/19619.
- After dismissing the third screen:
  - Fire a one-off POST request to ask the server to send an email to the user. This is covered in https://github.com/wordpress-mobile/WordPress-iOS/issues/19623.
  - Update the "migration-ready flag" to indicate that the migration process has been completed.
  - Show the success cards in JP.
  - **Optional due to deprioritization:** this would also be a great time to ensure that all the unneeded intermediate data is deleted from the App Group. Another option would be immediately after the import process completes. This is covered in https://github.com/wordpress-mobile/WordPress-iOS/issues/19658.

## Tasks

- [x] #19674
- [x] #19676
- [x] #19655
